draft-ietf-tls-tls13.md

```diff
@@ -2749,38 +2749,45 @@ servers MUST process the client's ClientHello and then immediately
 send the ServerHello, rather than waiting for the client's
 EndOfEarlyData message.
 
+#### Age and Lifetime {#age-lifetime}
+
+When a client create a NewSessionTicket message, the ticket_age is
+calculated as follows:
+
+       ticket_age = client_create_psk_time - client_recv_nst_time
+
+If "ticket_age < ticket_lifetime * 1000" is satisfied, the client can
+use the ticket. obfuscated_ticket_age is calculated as follows:
+
+       obfuscated_ticket_age = ticket_age_add - ticket_age
+
+When the server receives the NewSessionTicket message, ticket_age
+can be recovered as follows:
+
+       ticket_age = obfuscated_ticket_age + ticket_age_add
+
+The server MUST check "ticket_age < ticket_lifetime * 1000" where
+ticket_lifetime is retrieved from a local ticket database or decoded
+from the ticket.
+
 #### Replay Properties {#replay-time}
 
 As noted in {{zero-rtt-data}}, TLS provides a limited mechanism for
 replay protection for data sent by the client in the first flight.
+The server SHOULD limit the time over which the first flight might be
+replayed by checking:
 
-The "obfuscated_ticket_age" parameter in the client's
-"pre_shared_key" extension SHOULD be used by
-servers to limit the time over which the first flight might be
-replayed.  A server can store the time at which it sends a
-ticket to the client, or encode the time in the ticket.  Then, each
-time it receives an "pre_shared_key" extension, it can un-obfuscate the
-ticket age and compare it to the difference between the current time and
-the ticket creation time,
-to see if the value for the time elapsed sent by the client is
-consistent with the time since the ticket was issued.
-
-The ticket age (the value with "ticket_age_add" subtracted) provided by the
-client will be shorter than the
-actual time elapsed on the server by a single round trip time.  This
-difference is comprised of the delay in sending the NewSessionTicket
-message to the client, plus the time taken to send the ClientHello to
-the server.  For this reason, a server SHOULD measure the round trip
-time prior to sending the NewSessionTicket message and account for
-that in the value it saves.
-
-To properly validate the ticket age, a server needs to save at least two items:
+       (server_recv_psk_time - server_create_nst_time) - ticket_age < RTT + delta
```

Comment:
server_recv_psk_time needs to be explained/defined.
The left side might need to be the absolute value, but then again it might not, as sites are encouraged to use asymmetric replay windows a few paragraphs below.  So, this needs some rework.